### ISSUE TITLE
Build Issue on Alpine Linux

### ISSUE THREAD

**Issue by sodface** [TAAPArthur/navboard#1]
Hello, I successfully built both dtext-xcb and navboard on Alpine Linux. I've been looking for a simple OSK for a touchscreen car application and navboard seems great so far in my limited testing. I had a build error that I fixed with the below patch, though I'm not sure whether that's the best fix or not.

A question: what is meant by the "generic application launcher" part of the project description? Thanks!

```
--- ./xutil.c.orig
+++ ./xutil.c
@@ -191,7 +191,7 @@
             }
         }
     }
-    printf("Could not find %ld\n",targetSym);
+    printf("Could not find %ld\n",(long)targetSym);
     assert(0);
     return 0;
 }
```

**Comment by TAAPArthur** [comment]
Think the best fix would be to replace "%ld" with "%d". Your proposal wouldn't cause any problems either.

>A question: what is meant by the "generic application launcher" part of the project description? Thanks!

Yeah I need to actually add a readme. So the vision is that you press a button and then some arbitrary action happens. For the OSK it would send a key press. But it could just as easily spawn an arbitrary program; one of the default examples lets you choose the specific keyboard. I'll add more examples/be more descriptive in the readme that'll be added by the end of the week.

**Comment by sodface** [comment]
Thanks for the reply. And "simple" in my original post wasn't the right choice of words, I hope you didn't take that the wrong way, "lightweight" more like.

Do you plan to incorporate a fix for the build error I had or is that Alpine specific somehow and I should just keep my local patch and close this issue? Looking forward to the README update and I'll try and do some more testing this weekend.

//edit, thought I would also mention the following:

```
sodco:~$ navboard
Could not load font name /usr/share/fonts/TTF/LiberationMono-Regular.ttf

sodco:~$ sudo apk add ttf-liberation
(1/3) Installing font-liberation (2.1.5-r1)
(2/3) Installing font-liberation-sans-narrow (1.07.6-r1)
(3/3) Installing ttf-liberation (2.1.5-r1)

sodco:~$ navboard
Could not load font name /usr/share/fonts/TTF/LiberationMono-Regular.ttf

sodco:~$ sudo ln -s /usr/share/fonts/liberation/LiberationMono-Regular.ttf /usr/share/fonts/TTF/LiberationMono-Regular.ttf
sodco:~$ navboard
```

I see you can specify the default font at compile time but maybe worth mentioning in the README?

**Comment by TAAPArthur** [comment]
>I hope you didn't take that the wrong way, "lightweight" more like.

No worries. I didn't. I just don't know why I used "ld" which is for a long when the the data type is a normal length. A compiler may complain depending on the flags set (like -Werror), but I do plan to fix it. The fewer headaches for people the better.

For the font paths, I agree the current situation is annoying. As far as I'm aware there isn't a standard path for fonts. I'm thinking of detecting the "current" font path at compile time and/or being able to search for fonts at runtime

**Comment by TAAPArthur** [comment]
@sodface Are the build issues fixed now? Fair warning that you'll  have to reinstall dtext_xcb. I've made a lot of changes since we last talked.

Should have also fixed the issue with font paths. Now only a font name is specified and it should generally find the correct file.
Also added a README to better explain on what can be done

**Comment by sodface** [comment]
//edit 2, this line in the Makefile seems to clobber any previous CFLAGS assignments, I'm patching it out for now:

https://github.com/TAAPArthur/navboard/blob/a16b5aff6eb4091e62c36150593d50f2a7341c09/Makefile#L44

And deleted line 356 from dtext_xcb.h

```
dtext_xcb.h: In function 'dt_load_font':
dtext_xcb.h:356:10: error: unused variable 'buffer' [-Werror=unused-variable]
  356 |     char buffer[255];
      |          ^~~~~~
```
Building ok now on x86_64 and aarch64. Will continue testing, thanks!

Earlier comments follow:

//edit - disregard for now, I think this is operator error on my part, abuild is overriding the CFLAGS in the Makefile.

Thanks a lot for making the changes. Unfortunately, and keep in mind I only sort of know what I'm doing here, I'm running into this build error which I haven't been able to get by as of yet:

```
gcc -g -rdynamic -O0 -I /usr/include/freetype2 -shared -o libnavboard.so config.o util.o navboard.o xutil.o functions.o boards/mobile.o boards/numbers.o boards/qwerty.o -lX11 -lxcb -lxcb-ewmh -lxcb-icccm -lxcb-xtest -lxcb-render -lxcb-render-util -lfreetype
/usr/lib/gcc/x86_64-alpine-linux-musl/11.2.1/../../../../x86_64-alpine-linux-musl/bin/ld: boards/mobile.o: warning: relocation against `keys_mobile_symb' in read-only section `.text.startup'
/usr/lib/gcc/x86_64-alpine-linux-musl/11.2.1/../../../../x86_64-alpine-linux-musl/bin/ld: navboard.o: relocation R_X86_64_PC32 against symbol `boards' can not be used when making a shared object; recompile with -fPIC
/usr/lib/gcc/x86_64-alpine-linux-musl/11.2.1/../../../../x86_64-alpine-linux-musl/bin/ld: final link failed: bad value
collect2: error: ld returned 1 exit status
make: *** [Makefile:39: libnavboard.so] Error 1
make: *** Waiting for unfinished jobs....
>>> ERROR: navboard: build failed
```

I've been building navboard as a package using Alpine's build system since I already have a build server setup, so maybe there's a collision here with some flags or something?

**Comment by TAAPArthur** [comment]
My bad. The first one was a type-o. Meant for that line to only apply for tests.

dtext_xcb now has removed the extraneous line removed too

**Comment by sodface** [comment]
Rebuilt against latest sources and all is well. I'm not sure whether you are interested or not with distro packaging, but even though I've got working APKBUILDs going for Alpine here locally, which works great for me, I don't think I could submit them for addition to the official repos without some other tweaks. Mainly a versioned .so so I can split the package into a main and -dev package for the header files and unversioned .so symlink. Additionally, pulling down the dtext_xcb.h from a different repo works but I've never done that in an official package so I'm not sure if it would fly or not. Packaging against a git commit is ok but a release is better.

But anyway, locally, I get to set the rules so all bets are off and I'm pleased to have found navboard! Closing.
